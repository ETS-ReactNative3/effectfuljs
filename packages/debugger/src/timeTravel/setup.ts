import config from "../config";
config.timeTravel = true;
config.mutationObserver = true;
