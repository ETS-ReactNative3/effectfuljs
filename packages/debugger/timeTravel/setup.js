"use strict";

var _config = require("../config");

_config.default.timeTravel = true;
_config.default.mutationObserver = true;