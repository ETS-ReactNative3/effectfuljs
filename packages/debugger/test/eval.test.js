require("./setup");
const trace = require("./run").bind(null, true);
const D = require("../main");

test("eval tracing", function() {
  global.console = { log: jest.fn() };
  D.evalThunk(require("./__fixtures__/eval"));
  const mod = trace();
  mod.runEval();
  console.log("> eval", trace());
  expect(console.log.mock.calls).toMatchSnapshot();
});
