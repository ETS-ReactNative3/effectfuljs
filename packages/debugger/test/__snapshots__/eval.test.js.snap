// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eval tracing 1`] = `
Array [
  Array [
    "before eval",
    "g",
    "a",
    "b",
  ],
  Array [
    "hi",
    "g",
    "a",
    "b",
  ],
  Array [
    "there",
    "g",
    "abg",
    "b",
  ],
  Array [
    "eval",
    "abgbg",
  ],
  Array [
    "after eval",
    "g",
    "abg",
    "b",
  ],
  Array [
    "> eval",
    Array [
      "g",
      "abg",
      "b",
    ],
  ],
]
`;
